crypto/pem/pvkfmt.c

```diff
@@ -186,62 +186,61 @@ static unsigned int blob_length(unsigned bitlen, int isdss, int ispub)
 
 }
 
-static EVP_PKEY *do_b2i(const unsigned char **in, unsigned int length,
-                        int ispub)
+EVP_PKEY *ossl_b2i(const unsigned char **in, unsigned int length, int *ispub)
 {
     const unsigned char *p = *in;
     unsigned int bitlen, magic;
     int isdss;
-    if (ossl_do_blob_header(&p, length, &magic, &bitlen, &isdss, &ispub) <= 0) {
-        PEMerr(PEM_F_DO_B2I, PEM_R_KEYBLOB_HEADER_PARSE_ERROR);
+    if (ossl_do_blob_header(&p, length, &magic, &bitlen, &isdss, ispub) <= 0) {
```

Comment:
why does it need to pass a pointer to the header for ispub? Do you want it to be able to change from the passed in value for some reason?

Reply:
`ossl_do_blob_header` does already modify the value, if that value was -1...  if it was 0 or 1, it checks instead that the blob is what the caller expects.

Reply:
... and since `ossl_b2i` / `ossl_b2i_bio` are meant to be used for *discovery* (as seen by the use in `providers/implementations/serializer/deserializer_common.c`), it expects an `ispub` pointer from the caller and passes it on.